include/bluetooth/mesh/sensor.h

```diff
@@ -239,6 +262,116 @@ struct bt_mesh_sensor_format {
+	 *
+	 *  If @c val has a value that cannot be represented by the format,
+	 *  @c sensor_val will be set to the value clamped to the range
+	 *  supported by the format, and this function will return -ERANGE. This
+	 *  will clamp to "Greater than or equal to the maximum value" and "Less
+	 *  than or equal to the minimum value" if these are supported by the
+	 *  format.
+	 *
+	 *  If this function returns an error code other than -ERANGE,
+	 *  @c sensor_val is not modified.
+	 *
+	 *  @param[in]  format     Format to use when encoding the sensor value.
+	 *  @param[in]  val        The integer to convert.
+	 *  @param[out] sensor_val The resulting @ref bt_mesh_sensor_value.
+	 *
+	 *  @return 0 on success, (negative) error code otherwise.
+	 */
+	int (*const from_micro)(
+		const struct bt_mesh_sensor_format *format, int64_t val,
+		struct bt_mesh_sensor_value *sensor_val);
+
+	/** @brief Convert a @ref bt_mesh_sensor_value to a @c float.
+	 *
+	 *  If this function returns a status for which
+	 *  @ref bt_mesh_sensor_value_status_is_numeric returns false, @c val is
+	 *  not modified.
+	 *
+	 *  @param[in]  sensor_val The @ref bt_mesh_sensor_value to convert.
+	 *  @param[out] val        The resulting @c float.
+	 *
+	 *  @return The status of the conversion.
+	 */
+	enum bt_mesh_sensor_value_status (*const to_float)(
+		const struct bt_mesh_sensor_value *sensor_val, float *val);
+
+	/** @brief Convert a @c float to a @ref bt_mesh_sensor_value.
+	 *
+	 *  If @c val has a value that cannot be represented by the format,
+	 *  @c sensor_val will be set to the value clamped to the range
+	 *  supported by the format, and this function will return -ERANGE. This
+	 *  will clamp to "Greater than or equal to the maximum value" and "Less
+	 *  than or equal to the minimum value" if these are supported by the
+	 *  format.
+	 *
+	 *  If this function returns an error code other than -ERANGE,
+	 *  @c sensor_val is not modified.
+	 *
+	 *  @param[in]  format     Format to use when encoding the sensor value.
+	 *  @param[in]  val        The @c float to convert.
+	 *  @param[out] sensor_val The resulting @ref bt_mesh_sensor_value.
+	 *
+	 *  @return 0 on success, (negative) error code otherwise.
+	 */
+	int (*const from_float)(const struct bt_mesh_sensor_format *format,
+				float val,
+				struct bt_mesh_sensor_value *sensor_val);
+
+	/** @brief Convert a special @ref bt_mesh_sensor_value_status value to a
+	 *         @ref bt_mesh_sensor_value.
+	 *
+	 *  @param[in]  format     Format to use when encoding the sensor value.
+	 *  @param[in]  val        The @ref bt_mesh_sensor_value_status
+	 *                         to convert.
+	 *  @param[out] sensor_val The resulting @ref bt_mesh_sensor_value on
+	 *                         success. Undefined otherwise.
+	 *
+	 *  @return 0 on success, (negative) error code otherwise.
+	 */
+	int (*const from_special_status)(
+		const struct bt_mesh_sensor_format *format,
+		enum bt_mesh_sensor_value_status val,
```

Comment:
Nitpick: Change `val` to `status` to match the naming in `scalar_from_special_status` and `exp_1_1_from_special_status`.
Same in `bt_mesh_sensor_value_from_special_status`

Reply:
TBH, I'd rather do the other way around? Every other `bt_mesh_sensor_value_from_xyz` takes `val` and `sensor_val`

Reply:
To me, it does not make sense to call the parameter that is a `bt_mesh_sensor_value_status` is named "val(ue)", especially when the function also has `bt_mesh_sensor_value` as a parameter. In other instances where the enum is used you also declare it as "status".